src/meteodatalab/mars.py

```diff
@@ -62,19 +65,54 @@ class Point(typing.NamedTuple):
     lon: float
 
 
+@dc.dataclass(frozen=True)
+class Range:
+    start: int
+    end: int
+    step: int | None = None
+
```

Comment:
Is there validation to ensure the `Range` bounds are logically ordered, i.e., `start <= end`?
```suggestion

    @pydantic.model_validator(mode="after")
    def validate(self):
        if self.start > self.end:
            raise ValueError(f"Range start ({self.start}) must be <= end ({self.end})")
        if self.step is not None and self.step <= 0:
            raise ValueError(f"Range step ({self.step}) must be positive")
        return self
```

Reply:
Initially, I also thought about this validation rule but I backtracked thinking that potentially reversed ranges might be possible. I don't know if they are but we can also just restrict to forward ranges if you think that this makes sense.

Reply:
Right, maybe they allow it. Let's leave it on their side.